### PR TITLE
Refactor Digital Signature- LRQA-82784 | master

### PR DESCRIPTION
**Pull request by juliosilveira77**
https://liferay.atlassian.net/browse/LRQA-82784

### CODE REVIEW

**Comment by juliosilveira77** [comment]
ci:test:sf

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 1]
```diff
@@ -37,159 +39,65 @@ definition {
...
-			digitalSignatureAccountBaseUri = ${digitalSignatureAccountBaseUri},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_BASE_URI");
+		task ("Given the digital signature using the Inherit or Override strategy") {
+			DigitalSignature.enableDigitalSignature(siteStrategy = "Inherit or Override");
+		}
 
-		AssertElementPresent(
-			digitalSignatureIntegrationKey = ${digitalSignatureIntegrationKey},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_INTEGRATION_KEY");
+		task ("When you have it in Digital Signature") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:94]: Please, change this task to:`When the user goes to Digital Signature by portlet`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 2]
```diff
@@ -37,159 +39,65 @@ definition {
...
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_USER_ID");
-
-		AssertElementPresent(
-			digitalSignatureApiAccountId = ${digitalSignatureApiAccountId},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_ACCOUNT_ID");
-
-		AssertElementPresent(
-			digitalSignatureAccountBaseUri = ${digitalSignatureAccountBaseUri},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_BASE_URI");
+		task ("Given the digital signature using the Inherit or Override strategy") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:90]: Please, change this task to: `Given the digital signature credentials are set with Inherit or Override strategy`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 3]
```diff
@@ -37,159 +39,65 @@ definition {
...
-			digitalSignatureAccountBaseUri = ${digitalSignatureAccountBaseUri},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_BASE_URI");
+		task ("Given the digital signature using the Always Inherit strategy") {
+			DigitalSignature.enableDigitalSignature(siteStrategy = "Always Inherit");
+		}
 
-		AssertElementPresent(
-			digitalSignatureIntegrationKey = ${digitalSignatureIntegrationKey},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_INTEGRATION_KEY");
+		task ("When you have it in Digital Signature") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:78]: Please, change this task to:`When the user goes to Digital Signature by portlet`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 4]
```diff
@@ -37,159 +39,65 @@ definition {
...
-
-		AssertElementPresent(
-			digitalSignatureApiAccountId = ${digitalSignatureApiAccountId},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_ACCOUNT_ID");
-
-		AssertElementPresent(
-			digitalSignatureAccountBaseUri = ${digitalSignatureAccountBaseUri},
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_BASE_URI");
+		task ("Given the digital signature using the Always Inherit strategy") {
+			DigitalSignature.enableDigitalSignature(siteStrategy = "Always Inherit");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:75]: Please, change this task to: `Given the digital signature credentials are set with Always Inherit strategy`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 5]
```diff
@@ -37,159 +39,65 @@ definition {
...
+				dmDocumentFile = "Document_2.docx");
 
-		var dsStatus = "Sent";
+			Button.click(button = "Send");
+		}
 
-		AssertElementPresent(
-			dsStatus = ${dsStatus},
-			locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_STATUS");
+		task ("Then the digital signature can be enabled using the Always Override strategy") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:66]: Could you change this macro to: `Then the user can see that digital signature can be enabled using the Always Override strategy`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 6]
```diff
@@ -37,159 +39,65 @@ definition {
...
-			value1 = ${digitalSignatureRsaPrivateKey});
+			DigitalSignature.enableDigitalSignature();
 
-		SystemSettings.saveConfiguration();
+			SystemSettings.saveConfiguration();
 
-		ProductMenu.gotoPortlet(
-			category = "Content & Data",
-			portlet = "Digital Signature");
+			DigitalSignature.createEnvelope(
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:55]: Could you add a task before this macro: `And creates a Digital Signature envelope`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 7]
```diff
@@ -37,159 +39,65 @@ definition {
...
-		Type(
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_BASE_URI",
-			value1 = ${digitalSignatureAccountBaseUri});
+			SystemSettings.saveConfiguration();
+		}
 
-		Type(
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_INTEGRATION_KEY",
-			value1 = ${digitalSignatureIntegrationKey});
+		task ("When the credentials are placed and the digital signature is created") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:48]: Please, change this task to:` When the user set the credentials  by site settings`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase [hunk 8]
```diff
@@ -37,159 +39,65 @@ definition {
...
-		var digitalSignatureApiUsername = PropsUtil.get("digital.signature.api.username");
-		var digitalSignatureApiAccountId = PropsUtil.get("digital.signature.api.accountId");
-		var digitalSignatureAccountBaseUri = PropsUtil.get("digital.signature.account.base.uri");
-		var digitalSignatureIntegrationKey = PropsUtil.get("digital.signature.integration.key");
-		var digitalSignatureRsaPrivateKey = PropsUtil.get("digital.signature.rsa.private.key");
-
-		Type(
-			locator1 = "DigitalSignatureConfiguration#DIGITAL_SIGNATURE_USER_ID",
-			value1 = ${digitalSignatureApiUsername});
+		task ("Given the digital signature using the Always Override strategy") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:42]: Please, change this task to:`Given that the digital signature site strategy is set as 'Always Override'`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 9]
```diff
@@ -277,6 +326,12 @@ definition {
 			locator1 = "Checkbox#CHECKBOX_NAME");
 	}
 
+	macro siteSettingsStrategy {
+		Select(
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:330]: Could you change this macro name to: `selectSiteStrategy`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 10]
```diff
@@ -265,6 +302,18 @@ definition {
 			locator1 = "ListGroupItem#ITEM_TEXT");
 	}
 
+	macro navigateToComponent {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:305]: Could you change this macro name to: `goToDigitalSignatureByPortlet`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 11]
```diff
@@ -21,6 +21,41 @@ definition {
 			locator1 = "Header#H1_TITLE");
 	}
 
+	macro canDataPersistAfterEnabled {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:24]: Could you change this macro name to: `canSeeDataPersistAfterEnabled` and add this code:
```
var digitalSignatureApiUsername = PropsUtil.get("digital.signature.api.username");
		var digitalSignatureApiAccountId = PropsUtil.get("digital.signature.api.accountId");
		var digitalSignatureAccountBaseUri = PropsUtil.get("digital.signature.account.base.uri");
		var digitalSignatureIntegrationKey = PropsUtil.get("digital.signature.integration.key");
		var digitalSignatureRsaPrivateKey = PropsUtil.get("digital.signature.rsa.private.key");
```
due to it is needed to asset these values

**Review by MGabriellaS — COMMENTED** [review]
@juliosilveira77 Could you take a look when you get a chance?

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase (hunk 5 above)
Comment (juliosilveira77) [reply modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:66]: Task

Comment (MGabriellaS) [reply modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureSiteSettings.testcase:66]: oh sorry, my bad

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by MGabriellaS** [comment]
@juliosilveira77  thank you !! The code looks good to me!

**Comment by MGabriellaS** [comment]
ci:forward

**Pull request closed without merging**